apps/pipelines/models.py

```diff
@@ -95,16 +95,32 @@ def set_nodes(self, nodes: list[FlowNode]) -> None:
                 },
             )
 
+    def validate(self) -> dict:
+        """Validate the pipeline nodes and return a dictionary of errors"""
+        from apps.pipelines.nodes import nodes as pipeline_nodes
+
+        errors = {}
+
+        for node in self.node_set.all():
+            node_class = getattr(pipeline_nodes, node.type)
+            try:
+                node_class(**node.params)
```

Comment:
nit: use 'model_validate' to make this clearer
```suggestion
                node_class.model_validate(node.params)
```